### ISSUE TITLE
Invalid syntax error in multi-line f-string with format spec

### ISSUE THREAD

**Issue by dhruvmanila** [microsoft/pyright#6077]
**Describe the bug**

Basically the same bug as reported in CPython (https://github.com/python/cpython/issues/110259) exists in Pyright's lexer as well.

**Code or Screenshots**

```python
f"""__{
	a:d
}__"""
```

Running `pyright` on the above code gives:
```console
$ pyright ~/playground/ruff/fstring.py
/Users/dhruv/playground/ruff/fstring.py
  /Users/dhruv/playground/ruff/fstring.py:3:2 - error: String literal is unterminated
  /Users/dhruv/playground/ruff/fstring.py:2:5 - error: "x" is not defined (reportUndefinedVariable)
2 errors, 0 warnings, 0 informations 
```

**VS Code extension or command-line**

Running through the command-line using `1.1.329`

**Comment by erictraut** [comment]
I'm not able to repro the error with the code you've supplied above. Could you verify that your code sample is correct? For me, pyright emits another (correct) error indicating that `a` is an undefined symbol. This is in accordance with the python interpreter, which emits the same error. If I modify the code to remove that error and replace `a:d` with a valid expression, pyright emits no errors, nor does the Python interpreter.

```python
a = 1
d = 1

f"""__{
	a + d
}__"""
```

**Comment by erictraut** [comment]
Ah, I think you meant to use single quotes rather than triple quotes.

```python
f"__{
    x:d
}__"
```

Pyright's behavior matches that of the Python interpreter, so I consider it correct. I used the CPython test cases for PEP 701 to validate that pyright was working correctly here.

If the CPython devs ultimately determine that this is a bug in the Python interpreter, then I'll make the corresponding change in pyright. But for now, I consider this to be correct behavior.

**Comment by erictraut** [comment]
Oh, I just noticed that @pablogsal has already acknowledged that this is a bug in CPython and has submitted a PR to address it.

**Comment by erictraut** [comment]
This issue will be addressed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.330, which I just published. It will also be included in an upcoming release of pylance.
